.github/workflows/nextflow-tests.yml

```diff
@@ -73,11 +169,257 @@ jobs:
+                  ...updated_tests
+                ]
+              )
+            } catch (error) {
+              let prefix = "This is embarrassing..."
+
+              let comments = ${{ toJSON(steps.status.outputs.comments) }}
+
+              if (!comments.length || !comments[comments.length-1].includes(prefix)) {
+                let base_ref = ${{ toJSON(needs.discover.outputs.base_ref) }}
+                let branch_ref = ${{ toJSON(needs.discover.outputs.branch_ref) }}
+
+                github.rest.pulls.createReview({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  pull_number: ${{ toJSON(needs.discover.outputs.pr_number) }},
+                  commit_id: ${{ toJSON(needs.discover.outputs.review_sha) }},
+                  event: "REQUEST_CHANGES",
+                  body: `*Bleep bloop, I am a robot.*
+
+            ${prefix} the \`${ base_ref }\` branch has test changes that haven't been merged into _this_ branch, and I can't handle that. Please fix that and then I can fix the tests for you!
+
+            \`\`\`console
+            cd /path/to/your/repository/
+            git checkout ${ branch_ref }
+            git fetch origin
+            git merge origin/${ base_ref }
+            git push origin
+            \`\`\``
+                })
+              } else {
+                console.log("Review state not changed")
+              }
+
+              core.setFailed("Need to merge tests")
+            }
+
+      - if: ${{ steps.status.outputs.should-fix == 'true' }}
+        id: pushchanges
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: Autofix Nextflow configuration regression tests
+          branch: ${{ needs.discover.outputs.branch_ref }}
+          file_pattern: '**/configtest*.json'
+
+      - if: ${{ steps.status.outputs.should-fix == 'true' }}
+        name: Add comment about reviewing
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const timers = require('node:timers/promises')
+
+            let commit_id = ${{ toJSON(steps.pushchanges.outputs.commit_hash) }}
+
+            var review_details = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: ${{ toJSON(needs.discover.outputs.pr_number) }},
+              commit_id: commit_id,
+              event: "COMMENT",
+              body: `*Bleep bloop, I am a robot.*
+
+            I have updated all of the failing tests for you with ${ commit_id }. You must review my work before merging this pull request!`
+            }
+
+            // We need to trigger GitHub to actually compute the changes
+            // https://docs.github.com/en/rest/guides/using-the-rest-api-to-interact-with-your-git-database
+            for (let i=0; i<10; i++) {
+              // Wait the first time - it will always take some time
+              await timers.setTimeout(3000)
```

Comment:
Do we wait the first time because the auto-commit takes time to wrap up?

Reply:
It's not that auto-commit takes time to wrap up (it should have fully completed by the time this step started), it's that GitHub takes time to compute the mergeability and update the branch. Per the docs:

> A consuming API needs to explicitly request a pull request to create a test merge commit. A test merge commit is created when you view the pull request in the UI and the "Merge" button is displayed, or when you [get](https://docs.github.com/en/rest/pulls/pulls#get-a-pull-request), [create](https://docs.github.com/en/rest/pulls/pulls#create-a-pull-request), or [edit](https://docs.github.com/en/rest/pulls#update-a-pull-request) a pull request using the REST API. Without this request, the merge Git refs will fall out of date until the next time someone views the pull request.

Without waiting, the request creation fails with something like a "no such commit" error (I can't find an example). This loop just keeps checking until those things have updated.

Reply:
Got it. I'm not sure if I'm misreading something - 

This block is the PR request to create the test merge commit, right?

```javascript
var updated_pr = (await github.rest.pulls.get({
  owner: context.repo.owner,
  repo: context.repo.repo,
  pull_number: ${{ toJSON(needs.discover.outputs.pr_number) }}
})).data
```

If that's correct, why do we wait before the request?

Reply:
I... can't defend that, honestly. I feel like I had a reason, but by my own logic we shouldn't. Huh! Okay, I'll move the wait to the end of the block.

Reply:
I'm sure we will rediscover the reason by surprise some time in the future. :)

Reply:
Done with 8393a19.